src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -383,7 +383,7 @@ private async Task OnPartialValidationCompletedCallbackAsync(ValidationContext v
                 }
 
                 foreach (INetworkPeer peer in peersToBan)
-                    this.peerBanning.BanAndDisconnectPeer(peer.RemoteSocketEndpoint, validationContext.BanDurationSeconds, $"Invalid block received: {validationContext.Error.Message}");
+                    this.peerBanning.BanAndDisconnectPeerForDefaultInterval(peer.RemoteSocketEndpoint, $"Invalid block received: {validationContext.Error.Message}");
```

Comment:
This is wrong, leave the original as we need to be able to set duration on per rule basis

Reply:
Do we actually need banning time on per rule basis? 
The only way we ban is when the rule throws (so the context's error is set), and the rule throws when the block is invalid. It seems reasonable to me that no matter why the block is invalid we would like to ban the peer for the same time. 

But in case we don't- why rules should decide for how long the ban should happen? I think higher level logic should be responsible for that decision, not the rule itself.

Reply:
I think this discussion is outside of the scope of this PR. Bitcoin core does that, so possibly it is not a terrible idea. It would allow us to have better control about how strongly we ban peers for which problems.

Reply:
It's not a bad idea, I agree. 
But why rules get to decide the duration? It sounds like a high level logic to me comparing to just checking some block validity aspect. 

Also we might be validating our own blocks in which case ban duration is irrelevant (even though those are sanity checks).

Reply:
It might be for a discussion but it does not belong here. This issue and this PR is supposed to address something and is not supposed to do this. So if we want this, let's open an issue for discussion.

Reply:
Ok, makes sense. 
@zeptin can you please apply fixes?